### PR TITLE
[AAE-10103] language-menu storybook

### PR DESCRIPTION
**Pull request by DudaRobert**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [x] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://alfresco.atlassian.net/browse/AAE-10103

**What is the new behaviour?**
Added new stories for language-menu-component and language-picker-component

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/mock/language.service.mock.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { LanguageItem } from '../services/language-item.interface';
+
+@Injectable()
+export class LanguageServiceMock {
```
Comment (mauriziovitale) [inline lib/core/mock/language.service.mock.ts:23]: could you please introduce a new interface LanguageServiceInterface and implement it? In this way we are sure that the mock is in sync with the real service. 

Of course also LanguageService must implement the new interface LanguageServiceInterface
```suggestion
export class LanguageServiceMock implements LanguageServiceInterface {
```

File: lib/core/language-menu/language-picker.component.stories.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        <tr><td>CORE.METADATA.ACTIONS.SAVE</td><td>{{ 'CORE.METADATA.ACTIONS.SAVE' | translate }}</td></tr>
+        <tr><td>ADF.LANGUAGE</td><td>{{ 'ADF.LANGUAGE' | translate }}</td></tr>
+      </table>
+    `)
+];
+
+export const AsNestedMenu = Template.bind({});
+AsNestedMenu.decorators = [
+    componentWrapperDecorator(story => `
+      <style>
```
Comment (mauriziovitale) [inline lib/core/language-menu/language-picker.component.stories.ts:44]: I am not sure about this custom style, all the styles should come from the app itself, what do we do in the others stories?

File: lib/core/language-menu/language-picker.component.stories.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    component: LanguagePickerComponent,
+    title: 'Core/Components/Language Menu/Language Picker',
+    decorators: [
+        moduleMetadata({
+            imports: [CoreStoryModule, LanguageMenuModule, MatMenuModule, MatIconModule, TranslateModule]
+        })
+    ]
+} as Meta;
+
+const Template: Story<LanguageMenuComponent> = (args) => ({
```
Comment (mauriziovitale) [inline lib/core/language-menu/language-picker.component.stories.ts:20]: Template should be lowercase

Comment (mauriziovitale) [inline lib/core/language-menu/language-picker.component.stories.ts:20]: args: LanguageMenuComponent

File: lib/core/language-menu/language-picker.component.stories.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    props: args
+});
+
+export const Default = Template.bind({});
```
Comment (mauriziovitale) [inline lib/core/language-menu/language-picker.component.stories.ts:24]: Default, lowerCase

File: lib/core/language-menu/language-picker.component.stories.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+      <table>
+        <tr><th>Example key</th><th>Value</th></tr>
+        <tr><td>CORE.HOST_SETTINGS.TITLE</td><td>{{ 'CORE.HOST_SETTINGS.TITLE' | translate }}</td></tr>
+        <tr><td>CORE.METADATA.ACTIONS.SAVE</td><td>{{ 'CORE.METADATA.ACTIONS.SAVE' | translate }}</td></tr>
+        <tr><td>ADF.LANGUAGE</td><td>{{ 'ADF.LANGUAGE' | translate }}</td></tr>
+      </table>
+    `)
+];
+
+export const AsNestedMenu = Template.bind({});
```
Comment (mauriziovitale) [inline lib/core/language-menu/language-picker.component.stories.ts:41]: same as before

File: lib/core/mock/language.service.mock.ts (hunk 1 above)
Comment (DudaRobert) [reply lib/core/mock/language.service.mock.ts]: Added the Interface in new push, both LanguageService and LanguageServiceMock implement it.

File: lib/core/language-menu/language-picker.component.stories.ts (hunk 2 above)
Comment (DudaRobert) [reply lib/core/language-menu/language-picker.component.stories.ts]: Removed in new push.

File: lib/core/language-menu/language-picker.component.stories.ts (hunk 3 above)
Comment (DudaRobert) [reply lib/core/language-menu/language-picker.component.stories.ts]: fixed.

Comment (DudaRobert) [reply lib/core/language-menu/language-picker.component.stories.ts]: fixed.

File: lib/core/language-menu/language-picker.component.stories.ts (hunk 4 above)
Comment (DudaRobert) [reply lib/core/language-menu/language-picker.component.stories.ts]: fixed.

File: lib/core/language-menu/language-picker.component.stories.ts (hunk 5 above)
Comment (DudaRobert) [reply lib/core/language-menu/language-picker.component.stories.ts]: fixed.

File: lib/core/language-menu/language-menu.component.stories.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+import { CommonModule } from '@angular/common';
+import { MaterialModule } from '../material.module';
+import { TranslateModule } from '@ngx-translate/core';
+
+export default {
+    component: LanguageMenuComponent,
+    title: 'Core/Components/Language Menu/Language Menu',
+    decorators: [
+        moduleMetadata({
+            imports: [CoreStoryModule, CommonModule, MaterialModule, TranslateModule, LanguageMenuModule],
```
[the remark this answers is not in the document]
Comment (DudaRobert) [reply lib/core/language-menu/language-menu.component.stories.ts:34]: Removing those imports results in runtime errors in storybook 
Screenshot 2022-08-01 at 15 04 35

[the remark this answers is not in the document]
Comment (tomgny) [reply lib/core/language-menu/language-menu.component.stories.ts:34]: + Please, in `LanguageMenuModule` import only modules from angular material that are really needed

File: lib/core/language-menu/language-menu.component.stories.ts [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+            providers: [
+                { provide: LanguageService, useClass: LanguageServiceMock }
+            ]
+        })
+    ],
+    excludeStories: /.*Data$/
```
Comment (tomgny) [inline lib/core/language-menu/language-menu.component.stories.ts:40]: do we need that?

Comment (DudaRobert) [reply lib/core/language-menu/language-menu.component.stories.ts:40]: No actually, will remove that line in next push :)

File: lib/core/language-menu/language-picker.component.stories.ts [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+import { TranslateModule } from '@ngx-translate/core';
+import { LanguageService } from '../services/language.service';
+import { LanguageServiceMock } from '../mock/language.service.mock';
+
+export default {
+    component: LanguagePickerComponent,
+    title: 'Core/Components/Language Menu/Language Picker',
+    decorators: [
+        moduleMetadata({
+            imports: [CoreStoryModule, CommonModule, MaterialModule, TranslateModule, LanguageMenuModule],
```
Comment (tomgny) [inline lib/core/language-menu/language-picker.component.stories.ts:34]: same as above

File: lib/core/language-menu/language-picker.component.stories.ts [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+        changedLanguage: action('changedLanguage')
+    }
+});
+
+export const primary = languagePickerComponentTemplate.bind({});
+
+export const asNestedMenu = languagePickerComponentTemplate.bind({});
+asNestedMenu.decorators = [
+    componentWrapperDecorator(story => `
+      <button mat-icon-button class="dw-profile-menu" [matMenuTriggerFor]="menu">
```
Comment (tomgny) [inline lib/core/language-menu/language-picker.component.stories.ts:54]: do we need that class here? `class="dw-profile-menu"`

File: lib/core/mock/language.service.mock.ts [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { LanguageServiceInterface } from './../services/language.service.interface';
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { LanguageItem } from '../services/language-item.interface';
+
+@Injectable()
+export class LanguageServiceMock implements LanguageServiceInterface{
```
Comment (tomgny) [inline lib/core/mock/language.service.mock.ts:23]:
```suggestion
export class LanguageServiceMock implements LanguageServiceInterface {
```

File: lib/core/language-menu/language-picker.component.stories.ts (hunk 9 above)
Comment (DudaRobert) [reply lib/core/language-menu/language-picker.component.stories.ts:54]: Will be removed in next push

File: lib/core/mock/language.service.mock.ts [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+        {key: 'ar', label: 'العربية', direction: 'rtl'}
+    ]);
+
+    languages$ = this.languages.asObservable();
+
+    changeLanguage(_language: LanguageItem): void {
+    }
+
+    setLanguages(items: LanguageItem[]): void {
+        if (items && items.length > 0) {
```
Comment (tomgny) [inline lib/core/mock/language.service.mock.ts:51]:
```suggestion
        if (items?.length > 0) {
```

File: lib/core/services/language.service.ts [hunk 12]
```diff
@@ -15,14 +15,15 @@
...
+import { LanguageServiceInterface } from './language.service.interface';
 import { Injectable } from '@angular/core';
 import { BehaviorSubject } from 'rxjs';
 import { AppConfigService, AppConfigValues } from '../app-config/app-config.service';
 import { LanguageItem } from './language-item.interface';
 import { UserPreferencesService } from './user-preferences.service';
 
 @Injectable({providedIn: 'root'})
-export class LanguageService {
+export class LanguageService implements LanguageServiceInterface{
```
Comment (tomgny) [inline lib/core/services/language.service.ts:26]:
```suggestion
export class LanguageService implements LanguageServiceInterface {
```

File: lib/core/mock/language.service.mock.ts (hunk 11 above)
Comment (DudaRobert) [reply lib/core/mock/language.service.mock.ts:51]: This method is the same as in real service. Should I make the change there as well?

Comment (tomgny) [reply lib/core/mock/language.service.mock.ts:51]: yes please

**Review by tomgny — APPROVED** [review]

**Pull request merged**